tsdb/chunkenc/chunk.go

```diff
@@ -52,6 +52,16 @@ func IsValidEncoding(e Encoding) bool {
 	return e == EncXOR || e == EncHistogram || e == EncFloatHistogram
 }
 
+const (
+	// MaxBytesPerXORChunk is a soft cap for the size of an XOR chunk. It is a soft cap in that a final
+	// sample may be appended that exceeds the cap but no further samples will be appended to that chunk.
+	// This results in an effective hard cap of 1043 bytes (1024 bytes already in a chunk followed by a
+	// sample that takes the maximum size).
+	MaxBytesPerXORChunk = 1024
+	// TargetBytesPerNativeHistogramChunk sets a size target for each native histogram chunk.
```

Comment:
Maybe add that there are the same "soft cap" implications as above, just with a wider worst-case spread?